pkg/smb/nodeserver.go

```diff
@@ -501,15 +513,74 @@ func ensureKerberosCache(mountFlags []string, secrets map[string]string) (bool,
 		if err != nil {
 			return false, err
 		}
-		err = os.WriteFile(krb5CacheFileName, content, os.FileMode(0700))
+		// Write cache into volumeId-based filename, so it can be cleaned up later
+		volumeIDCacheFileName := volumeKerberosCacheName(volumeID)
+
+		var volumeIDCacheAbsolutePath = getKerberosFilePath(volumeIDCacheFileName)
+		err = os.WriteFile(volumeIDCacheAbsolutePath, content, os.FileMode(0700))
+		if err != nil {
+			return false, status.Error(codes.Internal, fmt.Sprintf("Couldn't write kerberos cache to file %s: %v", volumeIDCacheAbsolutePath, err))
+		}
+		err = os.Chown(volumeIDCacheAbsolutePath, credUID, credUID)
 		if err != nil {
-			return false, status.Error(codes.Internal, fmt.Sprintf("Couldn't write kerberos cache to file %s: %v", krb5CacheFileName, err))
+			return false, status.Error(codes.Internal, fmt.Sprintf("Couldn't chown kerberos cache %s to user %d: %v", volumeIDCacheAbsolutePath, credUID, err))
 		}
-		err = os.Chown(krb5CacheFileName, credUID, credUID)
+		// Delete target file if exists, it's safe to ignore errors.
+		_ = os.Remove(krb5CacheFileName)
```

Comment:
it's better print warning msg if remove file failed.

Reply:
Done